react/src/util/opReturn.ts

```diff
@@ -1,20 +1,109 @@
-export const OP_RETURN_PREFIX = 'paybutton$';
+import { lib, enc } from 'crypto-js';
 
-export function stringToHex(str: string): string {
-  return str
-    .split('')
-    .map(c => c.charCodeAt(0).toString(16).padStart(2, '0'))
-    .join('');
+// All the below variables are already encoded to HEX
+export const OP_RETURN_PREFIX_PUSHDATA = '04'; // \x04
+export const OP_RETURN_PREFIX = '50415900'; // PAY\x00
+export const VERSION = '00'; // \x00
+
+// 223 total bytes
+// - 1 from the OP_RETURN op code: '\x6a'
+// - 1 from the protocol pushdata: '\x04'
+// - 4 from the 4-byte procol prefix: 'PAY\x00'
+// - 1 for the version byte: '\x00'
+// - 1 from the 8-byte nonce pushdata: '\x08'
+// - 8 from the 8-byte nonce
+// - 2 from the maximum size for the data pushdata
+// = 205 available bytes
+export const USER_DATA_BYTES_LIMIT = 205;
+
+// Pushdata is self-describing up to 75 bytes, since 0x4c (76 in hex) is
+// a special OP code.
+const SINGLE_PUSHDATA_BYTE_LIMIT = 75;
+
+function prependNonceWithPushdata(hexString: string): string {
+  // 2 hex chars == 1 byte
+  const bytesQuantity = hexString.length / 2;
+  // We limit the nonce size to 75 bytes,
+  // since this is way more than necessary for security.
+  if (bytesQuantity > SINGLE_PUSHDATA_BYTE_LIMIT) {
+    throw new Error(
+      `Maximum ${SINGLE_PUSHDATA_BYTE_LIMIT} byte size exceeded for nonce: ${bytesQuantity}`,
+    );
+  }
+  const pushdata = bytesQuantity.toString(16).padStart(2, '0');
+  return `${pushdata}${hexString}`;
 }
 
-export function parseOpReturn(
-  opReturn: string | undefined,
-): string | undefined {
-  if (opReturn === undefined) return undefined;
-  const bytesQuantity = new Blob([opReturn]).size;
+function stringToHex(str: string): string {
```

Comment:
should decide on a spec for what kind of data will be in each push, then just always use that -- i.e. always ascii, always utf8, always hex to decimal number.

Reply:
Commented on top of the function